### PR TITLE
Feat: Project name validation

### PR DESCRIPTION
**Pull request by yahu1031**
Project name validations were added.

Close #499(as stale)

## Pre-launch Checklist

- [x] I read the [Contribute](https://docs.serverpod.dev/contribute) page and followed the process outlined there for submitting PRs.
- [x] This update contains only one single feature or bug fix and nothing else. (If you are submitting multiple fixes, please make multiple PRs.)
- [x] I read and followed the [Dart Style Guide](https://dart.dev/guides/language/effective-dart/style) and formatted the code with [dart format](https://dart.dev/tools/dart-format).
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`), and made sure that the documentation follows the same style as other Serverpod documentation. I checked spelling and grammar.
- [ ] I added new tests to check the change I am making.
- [x] All existing and new tests are passing.
- [x] Any breaking changes are documented below.

## Breaking changes

No breaking changes

### CODE REVIEW

File: tools/serverpod_cli/lib/src/util/string_validators.dart [hunk 1]
```diff
@@ -16,4 +16,7 @@ class StringValidators {
 
   static bool isValidTableIndexName(String name) =>
       _mixedSnakeCaseTester.hasMatch(name);
+
+  static bool isValidProjectName(String name) =>
+      RegExp(r'^[a-z]+(?:[0-9a-z]+|_[0-9a-z]+)*$').hasMatch(name);
```
Comment (Isakdl) [inline tools/serverpod_cli/lib/src/util/string_validators.dart:21]: Please add a description on the valid format of the regex, could you also add some unit tests to validate these rules.

**Review by Isakdl — CHANGES_REQUESTED** [review]
Thank you for the PR! This is a needed change!

Before we merge could you add some unit tests to validate the validation function. It would also be cool to have an integration test to validate the behavior when creating a project.

File: tools/serverpod_cli/lib/src/util/string_validators.dart (hunk 1 above)
Comment (yahu1031) [reply tools/serverpod_cli/lib/src/util/string_validators.dart:21]: BTW can I get the unit test folder path.

Comment (Isakdl) [reply tools/serverpod_cli/lib/src/util/string_validators.dart:21]: There is a `test` folder in the serverpod_cli project!

File: tools/serverpod_cli/bin/serverpod_cli.dart [hunk 2]
```diff
@@ -239,12 +239,9 @@ Future<void> _main(List<String> args) async {
         return;
       }
 
-      var re = RegExp(r'^[a-z0-9_]+$');
-      if (results.arguments.length > 1 && re.hasMatch(name)) {
-        await performCreate(name, verbose, template, force);
-        _analytics.cleanUp();
-        return;
-      }
+      await performCreate(name, verbose, template, force);
```
Comment (yahu1031) [inline tools/serverpod_cli/bin/serverpod_cli.dart:242]: Removed the regex check cause it was added internally in the `performCreate` function

**Comment by Isakdl** [comment]
LGTM! :)

**Pull request merged**